compilator/compilator_linux.go

```diff
@@ -0,0 +1,85 @@
+package compilator
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+
+	"github.com/fatih/color"
+	"github.com/hpcloud/fissile/docker"
+	"github.com/hpcloud/fissile/model"
+	"github.com/hpcloud/fissile/scripts/compilation"
+)
+
+func (c *Compilator) compilePackageInChroot(pkg *model.Package) (err error) {
+	// Prepare input dir (package plus deps)
+	if err := c.createCompilationDirStructure(pkg); err != nil {
+		return fmt.Errorf("failed to create directory: %s", err)
+	}
+
+	if err := c.copyDependencies(pkg); err != nil {
+		return fmt.Errorf("failed to copy dependencies: %s", err)
+	}
+
+	// Generate a compilation script
+	targetScriptName := "compile.sh"
+	hostScriptPath := filepath.Join(pkg.GetTargetPackageSourcesDir(c.hostWorkDir), targetScriptName)
+	if err := compilation.SaveScript(c.baseType, compilation.CompilationScript, hostScriptPath); err != nil {
+		return fmt.Errorf("failed to copy compilation script: %s", err)
+	}
+
+	// Extract package
+	extractDir := c.getSourcePackageDir(pkg)
+	if _, err := pkg.Extract(extractDir); err != nil {
+		return fmt.Errorf("faile to extract package: %s", err)
+	}
+
+	// in-memory buffer of the log
+	log := new(bytes.Buffer)
+
+	stdoutWriter := docker.NewFormattingWriter(
+		log,
+		func(line string) string {
+			return color.GreenString("compilation-%s > %s", color.MagentaString("%s", pkg.Name), color.WhiteString("%s", line))
+		},
+	)
+	stderrWriter := docker.NewFormattingWriter(
```

Comment:
Might be time to move `NewFormattingWriter` out of the docker package? Seems a bit odd is all.